xpublish/routers/xyz.py

```diff
@@ -0,0 +1,34 @@
+import xarray as xr
+from fastapi import APIRouter, Depends, Response
+
+from xpublish.dependencies import get_dataset
+from xpublish.utils.ows import (
+    get_image_datashader,
+    get_bounds,
+    LayerOptionsMixin,
+    get_tiles,
+)
+
+
+class XYZRouter(APIRouter, LayerOptionsMixin):
+    pass
+
+
+xyz_router = XYZRouter()
+
+
+@xyz_router.get("/tiles/{layer}/{time}/{z}/{x}/{y}")
```

Comment:
Perhaps we could make less assumptions here about the structure of the dataset. I'd rather see something like `/tiles/{var}/{z}/{x}/{y}` and allow setting `time` or any other extra dimension(s) as a query parameter.

Allowing flexible image formats `/tiles/{var}/{z}/{x}/{y}.{format}` would be great.

(If later xarray supports multiscale datasets https://github.com/pydata/xarray/issues/4118 https://github.com/pydata/xarray/issues/5376 it will be nice to have `/tiles/{var}/{z}/{x}/{y}@{scale}x` too)

Reply:
@benbovy 

> Perhaps we could make less assumptions here about the structure of the dataset. I'd rather see something like `/tiles/{var}/{z}/{x}/{y}` and allow setting `time` or any other extra dimension(s) as a query parameter.

yes it makes sense to be minimalist in the assumption about data structure. 
In general the get_tile operation should be used by every tile services (XYZ, WMTS...). Different tile services can then implement a certain logic on how other dimensions  (time included) can be managed.
 
These are some references about  time dimension in WMTS:

- https://wiki.earthdata.nasa.gov/display/GIBS/GIBS+API+for+Developers#GIBSAPIforDevelopers-OGCWebMapTileService(WMTS)

- http://demo.geo-solutions.it/share/wmts-multidim/wmts_multidim.html

- http://docs.opengeospatial.org/per/16-042r1.html

Reply:
@benbovy which other image formats do you think it should support?